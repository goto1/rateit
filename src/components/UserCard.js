import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import UserItemContents from "./UserItemContents";
import RatingCategoriesList from "./RatingCategoriesList";
import HorizontalRule from "./HorizontalRule";
import {
  AccordionContent,
  AccordionItem,
  AccordionToggle,
  Card,
  CardContent,
  CardFooter,
  Icon
} from "./f7";

const Row = styled.div`display: flex;`;

export const Button = styled.button.attrs({
  type: "button"
})`
  border: none;
  background: inherit;
  font-size: 13px;
  padding: 0;
  flex-basis: ${props => props.flexBasis || "100%"};
  text-align: ${props => props.textAlign};
  &:focus { outline: none; }
  i {
    font-size: 17px;
    margin: ${props => props.iconMargin};
  }
`;

const AccordionToggleStyled = styled(AccordionToggle)`
  flex-basis: 100%;
`;

const StyledLink = styled.a`
  width: 100%;
  text-align: right;
`;

const Actions = ({ aggregateRatings, userId }) =>
  <AccordionItem>
    <Row>
      <AccordionToggleStyled>
        <Button textAlign="left" iconMargin="0 0 0 5px">
          Quick Overview
          <Icon material="arrow_downward" />
        </Button>
      </AccordionToggleStyled>
      <StyledLink href={`/user/${userId}`}>
        <Button textAlign="right" iconMargin="0 5px 0 0">
          <Icon material="person" />
          See Profile
        </Button>
      </StyledLink>
    </Row>
    <AccordionContent>
      <RatingCategoriesList
        ratings={aggregateRatings}
        hrColors={{ colorOne: "#E5E5E5", colorTwo: "#0B7EFF" }}
      />
    </AccordionContent>
  </AccordionItem>;

Actions.propTypes = {
  aggregateRatings: PropTypes.array.isRequired
};

const CardFooterStyled = styled(CardFooter)`
  flex-direction: row-reverse;
`;

const UserCard = props => {
  const { id, aggregateRatings, removeUserFromBookmarks } = props;
  return (
    <Card>
      <CardContent>
        <UserItemContents {...props} />
        <HorizontalRule
          margin="6px auto"
          width="90%"
          colorOne="#e5e5e5"
          colorTwo="#0b7eff"
        />
        <Actions aggregateRatings={aggregateRatings} userId={id} />
      </CardContent>
      <CardFooterStyled>
        <Button
          onClick={() => removeUserFromBookmarks(id)}
          textAlign="right"
          iconMargin="0 0 0 7.5px"
          flexBasis="75px"
        >
          Saved
          <Icon material="bookmark" />
        </Button>
      </CardFooterStyled>
    </Card>
  );
};

UserCard.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf(["student", "professor"]),
  schools: PropTypes.array.isRequired,
  overallRating: PropTypes.number.isRequired,
  userRatings: PropTypes.array.isRequired,
  aggregateRatings: PropTypes.array.isRequired,
  removeUserFromBookmarks: PropTypes.func.isRequired
};

export default UserCard;
