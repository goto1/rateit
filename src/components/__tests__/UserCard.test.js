import React from "react";
import UserCard, { Button } from "../UserCard";
import renderer from "react-test-renderer";
import { shallow, render, mount } from "enzyme";

const props = {
  id: "UArjrbxWHX",
  name: "Randall Rivera",
  type: "student",
  schools: [
    {
      id: "bruHy",
      name: "New Jersey Institute of Technology, Newark",
      abbreviation: "NJIT"
    },
    {
      id: "5y7fh",
      name: "Rutgers University, New Brunswick",
      abbreviation: "RU"
    }
  ],
  overallRating: 4.5,
  aggregateRatings: [
    { description: "Attends Group Meetings", rating: 3 },
    { description: "Contributes to Discussions", rating: 5 }
  ],
  userRatings: [
    {
      id: "gXv6EmCRJ",
      userId: "UArjrbxWHX",
      authorId: "TxjkPq46BG",
      authorName: "EagerPig",
      date: "1499183677321",
      comment:
        "I have no issues with the guy, however, he was sometimes unavailable due to his job.",
      likes: 9,
      dislikes: 0,
      aggregateRating: 4.5,
      recommendUser: true,
      individualRatings: [
        { id: "DRFJY9jd", rating: 3 },
        { id: "j84WAR77", rating: 5 }
      ]
    }
  ]
};

describe("UserCard component", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<UserCard {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it("should render the user name", () => {
    const wrapper = render(<UserCard {...props} />);
    const actual = wrapper.html().includes("Randall Rivera");
    const expected = true;
    expect(actual).toBe(expected);
  });

  it("should render the correct amount of user ratings", () => {
    const wrapper = render(<UserCard {...props} />);
    const actual = wrapper.html().includes("<span>1</span> ratings");
    const expected = true;
    expect(actual).toBe(expected);
  });
});
