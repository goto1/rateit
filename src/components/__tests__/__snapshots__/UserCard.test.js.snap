// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component should render correctly 1`] = `
.cqISHI {
  width: 95px;
}

.kXEDlD {
  font-size: 19px;
  color: #FEBA72;
  width: 19px;
  padding-bottom: 2px;
}

.fgWKlg {
  font-size: 15px;
  font-weight: 500;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  color: #497B3C;
}

.dWXrjy {
  border-radius: 5px;
  background-color: #0b7eff;
  color: #fff;
  -webkit-text-align: center;
  text-align: center;
  font-weight: 300;
  font-size: 15px;
  width: 32.5px;
  padding: 1px;
  margin: 0;
}

.kVNQGV {
  color: rgb(0, 0, 0);
  font-size: 14px;
  font-weight: 500;
  -webkit-letter-spacing: .5px;
  -moz-letter-spacing: .5px;
  -ms-letter-spacing: .5px;
  letter-spacing: .5px;
}

.jyiPTP {
  color: #8e8e93;
  font-size: 14px;
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  margin-top: 2px;
}

.jyiPTP span {
  font-weight: 600;
}

.gNYAYU {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
}

.jfGaMs {
  width: 100%;
}

.eTkWIx {
  -webkit-letter-spacing: 2px;
  -moz-letter-spacing: 2px;
  -ms-letter-spacing: 2px;
  letter-spacing: 2px;
}

.jJZhyj {
  font-weight: 500;
  color: #DEB21C;
}

.bJzcFu {
  font-weight: 500;
  color: #A1BA37;
}

.jFINBM {
  border: 0px;
  height: 1px;
  width: 90%;
  margin: 6px auto;
  background-image: linear-gradient(    to left,    #e5e5e5,    #0b7eff,    #e5e5e5  );
}

.eDYRKH {
  border: 0px;
  height: 1px;
  width: 70%;
  margin: 1px auto;
  background-image: linear-gradient(    to left,    #E5E5E5,    #0B7EFF,    #E5E5E5  );
}

.dJwEYn {
  -webkit-flex-grow: 1;
  -ms-flex-grow: 1;
  flex-grow: 1;
  padding-top: 4px;
  color: #000;
}

.eBUYwb {
  -webkit-flex-basis: 22px;
  -ms-flex-basis: 22px;
  flex-basis: 22px;
  font-size: 19px;
  padding-top: 4px;
}

.eMhhnH {
  font-size: 14px;
}

.eMhhnH span {
  font-size: 17px;
}

.zQFoY {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1px;
}

.dorDaK {
  margin-top: 7.5px;
}

.jnGuGo {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.fbgClP {
  border: none;
  background: inherit;
  font-size: 13px;
  padding: 0;
  -webkit-flex-basis: 100%;
  -ms-flex-basis: 100%;
  flex-basis: 100%;
  -webkit-text-align: left;
  text-align: left;
}

.fbgClP:focus {
  outline: none;
}

.fbgClP i {
  font-size: 17px;
  margin: 0 0 0 5px;
}

.iXErJR {
  border: none;
  background: inherit;
  font-size: 13px;
  padding: 0;
  -webkit-flex-basis: 100%;
  -ms-flex-basis: 100%;
  flex-basis: 100%;
  -webkit-text-align: right;
  text-align: right;
}

.iXErJR:focus {
  outline: none;
}

.iXErJR i {
  font-size: 17px;
  margin: 0 5px 0 0;
}

.boyVqA {
  border: none;
  background: inherit;
  font-size: 13px;
  padding: 0;
  -webkit-flex-basis: 75px;
  -ms-flex-basis: 75px;
  flex-basis: 75px;
  -webkit-text-align: right;
  text-align: right;
}

.boyVqA:focus {
  outline: none;
}

.boyVqA i {
  font-size: 17px;
  margin: 0 0 0 7.5px;
}

.fQTHsC {
  -webkit-flex-basis: 100%;
  -ms-flex-basis: 100%;
  flex-basis: 100%;
}

.bbaXTp {
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

<div
  className="card"
>
  <div
    className="card-content"
  >
    <div
      className="card-content-inner"
    >
      <div
        className="sc-chPdSV jfGaMs"
      >
        <div
          className="sc-kAzzGY gNYAYU"
        >
          <div
            className="username sc-jTzLTM fgWKlg"
            color="#497B3C"
          >
            Randall Rivera
          </div>
          <div
            className="sc-fjdhpX dWXrjy"
          >
            4.5
          </div>
        </div>
        <div
          className="sc-kAzzGY gNYAYU"
        >
          <div
            className="sc-jzJRlG kVNQGV"
          >
            Student
             at
             
            NJIT, RU
          </div>
        </div>
        <div
          className="sc-kAzzGY gNYAYU"
        >
          <div
            className="sc-cSHVUG jyiPTP"
          >
            <span>
              1
            </span>
             ratings
          </div>
          <div
            className="sc-gqjmRU cqISHI"
          >
            <i
              className="material-icons icon sc-VigVT kXEDlD"
            >
              star
            </i>
            <i
              className="material-icons icon sc-VigVT kXEDlD"
            >
              star
            </i>
            <i
              className="material-icons icon sc-VigVT kXEDlD"
            >
              star
            </i>
            <i
              className="material-icons icon sc-VigVT kXEDlD"
            >
              star
            </i>
            <i
              className="material-icons icon sc-VigVT kXEDlD"
            >
              star_half
            </i>
          </div>
        </div>
      </div>
      <hr
        className="sc-kpOJdX jFINBM"
        width="90%"
      />
      <div
        className="accordion-item"
      >
        <div
          className="sc-kEYyzF jnGuGo"
        >
          <div
            className="accordion-item-toggle sc-iAyFgw fQTHsC"
          >
            <button
              className="sc-kkGfuU fbgClP"
              onClick={[Function]}
              type="button"
            >
              Quick Overview
              <i
                className="material-icons icon"
              >
                arrow_downward
              </i>
            </button>
          </div>
          <button
            className="sc-kkGfuU iXErJR"
            onClick={[Function]}
            type="button"
          >
            <i
              className="material-icons icon"
            >
              person
            </i>
            See Profile
          </button>
        </div>
        <div
          className="accordion-item-content"
        >
          <div
            className="sc-hMqMXs dorDaK"
          >
            <div>
              <div
                className="sc-eNQAEJ zQFoY"
              >
                <i
                  className="material-icons icon sc-ckVGcZ eBUYwb"
                >
                  forward
                </i>
                <div
                  className="sc-dxgOiQ dJwEYn"
                >
                  Attends Group Meetings
                </div>
                <div
                  className="sc-jKJlTe eMhhnH sc-kgoBCf eTkWIx"
                >
                  <span
                    className="sc-kGXeez jJZhyj"
                    color="#DEB21C"
                  >
                    3
                  </span>
                  /5
                </div>
              </div>
              <hr
                className="sc-kpOJdX eDYRKH"
                width="70%"
              />
            </div>
            <div>
              <div
                className="sc-eNQAEJ zQFoY"
              >
                <i
                  className="material-icons icon sc-ckVGcZ eBUYwb"
                >
                  forward
                </i>
                <div
                  className="sc-dxgOiQ dJwEYn"
                >
                  Contributes to Discussions
                </div>
                <div
                  className="sc-jKJlTe eMhhnH sc-kgoBCf eTkWIx"
                >
                  <span
                    className="sc-kGXeez bJzcFu"
                    color="#A1BA37"
                  >
                    5
                  </span>
                  /5
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="card-footer sc-hSdWYo bbaXTp"
  >
    <button
      className="sc-kkGfuU boyVqA"
      onClick={[Function]}
      type="button"
    >
      Saved
      <i
        className="material-icons icon"
      >
        bookmark
      </i>
    </button>
  </div>
</div>
`;
